### PR TITLE
Make parseSQLFlowStmt return error location

### LINKED ISSUE #1478: parseSQLFlowStmt must clear out global variable before parsing
Right here, 

https://github.com/sql-machine-learning/sqlflow/blob/36066f128f99acc23e47f6fb93e9d5a6081fad5e/pkg/parser/extended_syntax_parser.y#L452-L454

we must clear the global variable `parseResult`.  Otherwise, `parseSQLFlowStmt` might return the result from the previous call.

### LINKED ISSUE #1479: Make extended syntax parser able to return error location
To remove `SplitMultipleSQL` as discussed in https://github.com/sql-machine-learning/sqlflow/issues/1430#issuecomment-565889881, we need to make the extended syntax parser return error location as what external parsers do.  After that, we could rewrite `parser.Parse` and make it independent to the ad-hoc `SplitMultipleSQL`.

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fix https://github.com/sql-machine-learning/sqlflow/issues/1478

Fix https://github.com/sql-machine-learning/sqlflow/issues/1479

### CODE REVIEW

File: pkg/parser/extended_syntax_parser.y [hunk 1]
```diff
@@ -436,12 +436,12 @@ func (s StandardSelect) String() string {
...
-func parseSQLFlowStmt(s string) (r *SQLFlowSelectStmt, e error) {
+func parseSQLFlowStmt(s string) (r *SQLFlowSelectStmt, idx int, e error) {
 	defer func() {
 		if r := recover(); r != nil {
-			var ok bool
-			e, ok = r.(error)
-			if !ok {
+			if err, ok := r.(error); ok {
+			        e = err
+                        } else {
```
Comment (tonyyang-svail) [inline pkg/parser/extended_syntax_parser.y:444]: indentation.

File: pkg/parser/extended_syntax_parser.y [hunk 2]
```diff
@@ -450,6 +450,12 @@ func parseSQLFlowStmt(s string) (r *SQLFlowSelectStmt, e error) {
 	mu.Lock()
 	defer mu.Unlock()
 
-	extendedSyntaxParse(newLexer(s))  // extendedSyntaxParse is auto generated.
-	return parseResult, nil
+	parseResult = nil // Important! Clear out result from previous call.
+        lex := newLexer(s)
```
Comment (tonyyang-svail) [inline pkg/parser/extended_syntax_parser.y:454]: indentation.

Comment (typhoonzero) [inline pkg/parser/extended_syntax_parser.y:454]: Indent error?

File: pkg/parser/extended_syntax_parser_test.go [hunk 3]
```diff
@@ -240,18 +248,34 @@ func TestExtendedSyntaxParseSelectStarAndPrint(t *testing.T) {
...
+		a.Nil(r)
+		a.Equal(0, idx)
+		a.Error(e)
+	}
+	{
+		r, idx, e := parseSQLFlowStmt(`   DROP TABLE TO PREDICT`)
+		a.Nil(r)
+		a.Equal(3, idx) // right before DROP as there was an error.
+		a.Error(e)
+	}
```
Comment (tonyyang-svail) [inline pkg/parser/extended_syntax_parser_test.go:264]: Maybe add more tests on statements like
1. Missing WITH clause `select ... to train dnn label class ...`
1. Missing semicolon in the with clause `select ... to train dnn with a = b c = d...`.
1. Addition semicolon in the with clause `select ... to train dnn with a = b, c=d, column ...`

Users often make these mistakes. And the error message from the parser can be really helpful.

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

File: pkg/parser/extended_syntax_parser.y (hunk 2 above)
Comment (wangkuiyi) [reply pkg/parser/extended_syntax_parser.y:454]: Yes. Go fmt doesn't handle .y files. I will fire another PR with tools that can auto-indent .y files.

**Pull request merged**